### PR TITLE
STYLE: fix line length check of flake8

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
I think I made a small mistake in https://github.com/pandas-dev/pandas/pull/27076 in adding the E501 code (line length) to the ignore list, as setting the flake8 max line length to 88 should have been sufficient.

Black does not check or correct line length in comments or multiline strings (eg docstrings), so this still need to be catched by flake8

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
But, this gives some interesting cases, it seems not always easy to reconcile both ...

File: pandas/core/groupby/grouper.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def __init__(
             if self.name is None:
                 self.name = index.names[level]
 
-            self.grouper, self._labels, self._group_index = index._get_grouper_for_level(
+            self.grouper, self._labels, self._group_index = index._get_grouper_for_level(  # noqa
```
Comment (jorisvandenbossche) [inline pandas/core/groupby/grouper.py]: this is a case where black produces too long line lengths (the original code before black was using `\` after `=` for line continuation, but checked, and black always removes this when adding it back.

For now just adding `# noqa` for this and similar cases (long term the code could be refactored a bit too avoid this, but now just fixing flake8)

Comment (WillAyd) [reply pandas/core/groupby/grouper.py]: Makes sense to me!

Comment (WillAyd) [reply pandas/core/groupby/grouper.py]: Actually can you add `# noqa: E501` to these? I suppose makes sense to only filter the line length error here

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Pull request merged**